src/Dao/PlagiabotDao.php

```diff
@@ -68,6 +67,11 @@ public function getPlagiarismRecords( $n = 50, $options ) {
 					$filters[] = "status IS NULL";
 					break;
 			}
+			// search filters
+			if ( $searchCriteria == 'page' && $searchText ) {
+				$searchText = filter_var( $searchText, FILTER_SANITIZE_STRING );
```

Comment:
Going off of http://php.net/manual/en/filter.filters.sanitize.php it looks like `FILTER_SANITIZE_STRING` won't prevent SQL injection, instead just removing tags and encoding.

Did the bind keys thing not work? https://github.com/wikimedia/CopyPatrol/pull/49#discussion_r144409785

Reply:
Yeah, bind doesn't work. According to docs, the placeholder cannot have quotes around it. It took me a while to figure out why everything was broken after I wrote that code and apparently this is somewhat obscure. But I cannot, for example, do "... LIKE '%:searchText%'"
I don't know what's a better way to do this. 🤔

Reply:
I guess what we basically want is `mysql_real_escape_string()` but Slim abstracts it so we can't access the mysql connection object in order to be able to use that...

Reply:
Seems like you would want to use mysqli_real_escape_string() here instead of filter_var().

Reply:
Can't - we don't have the connection object.
I think http://php.net/manual/en/function.mysql-escape-string.php will do the trick. It's deprecated (and later removed) but works for our case.

Reply:
@Niharika29: If you apply mysqli_real_escape_string() before it gets passed to Slim, I think it should work...
```
$searchText = mysqli_real_escape_string( $searchText );
$filters[] = "page_title LIKE '%" . $searchText . "%'";
```
Note mysqli rather than mysql, BTW.

Reply:
I found this https://stackoverflow.com/a/12059603/604142

I would try something like:
```php
$filters[] = "page_title LIKE CONCAT('%', :searchtext, '%');
$preparedParams['searchtext'] = $searchText;
```

Reply:
@kaldari - It won't. It needs a connection. See http://php.net/manual/en/mysqli.real-escape-string.php

@MusikAnimal - Tried that. Didn't work. I've killed quite some time around this bug now. :P

Reply:
This might be a bug in Slim. `CONCAT` should work, e.g. try `CONCAT('%', 'Good', '%')`. I think it is inferring the wrong data type.

Reply:
Nvm it is working for me... I tried this (also replacing spaces with underscores since that is how they are stored in the db):
```
$filters[] = "page_title LIKE CONCAT('%', :searchtext, '%')";
$preparedParams['searchtext'] = str_replace(' ', '_', $searchText);
```